src/error.rs

```diff
@@ -124,6 +124,9 @@ pub enum ErrorKind {
     #[error(display = "{}", _0)]
     Io(#[error(source)] std::io::Error),
 
+    #[error(display = "No DNS results for domain {}", _0)]
+    NoDnsResults(StreamAddress),
```

Comment:
Could we lump the `DnsName` error in with this one?

Reply:
That would be tricky; `DnsName` wraps an external error (from `webpki`), so we'd have to turn it into an enum itself, which doesn't seem much better than just putting another variant here. Note also that `DnsName` [signifies that the domain name a user passed us isn't syntactically valid](https://docs.rs/webpki/0.21.2/webpki/struct.InvalidDNSNameError.html), which is a bit different from a DNS query returning no results.

Reply:
Perhaps we should rename the `DnsName` case to better indicate that now that we have a similarly named error case, e.g. `InvalidDnsName`. This would also match up with `InvalidHostname` nicely.

Reply:
I'm not sure I understand why users would confuse `DnsName` with `NoDnsResults`. `NoDnsResults` seems more similar to `DnsResolve` than `DnsName` to me. That said, I'm not opposed to the name `InvalidDnsName`, but that would be a breaking change, so we probably want to be pretty certain that we want to change error names before doing it. In particular, if there's another name for `NoDnsResults` that you think would improve things instead, that would be a preferable solution since it's not an existing error type yet.

Reply:
The driver is currently in alpha; breaking changes should not be a concern whatsoever. Also, I have no issue with `NoDnsResults` since it is named very specifically. `DnsName` is too generic in my opinion and should be changed to match `InvalidHostname`.

Reply:
I'm deferring to both of you on this one, but I agree that we should feel comfortable making breaking changes until 1.0 comes out. If we feel like `InvalidDnsName` is going to help a user better understand what's going on, then we should make that change.

Reply:
How about we file another ticket to change it to `InvalidDnsName` and then schedule it? It's not really related to the work this ticket is doing at all, so I think it makes sense to follow up with a separate commit right after rather than bundle it in here.

Reply:
that's fine with me.

Reply:
Filed https://jira.mongodb.org/browse/RUST-295